This patch makes it possible to set a custom image magic.

--- a/tools/mkimage.c
+++ b/tools/mkimage.c
@@ -20,6 +20,7 @@ static struct image_tool_params params =
 	.arch = IH_ARCH_PPC,
 	.type = IH_TYPE_KERNEL,
 	.comp = IH_COMP_GZIP,
+	.magic = IH_MAGIC,
 	.dtc = MKIMAGE_DEFAULT_DTC_OPTIONS,
 	.imagename = "",
 	.imagename2 = "",
@@ -76,11 +77,12 @@ static void usage(const char *msg)
 			 "          -l ==> list image header information\n",
 		params.cmdname);
 	fprintf(stderr,
-		"       %s [-x] -A arch -O os -T type -C comp -a addr -e ep -n name -d data_file[:data_file...] image\n"
+		"       %s [-x] -A arch -O os -T type -C comp -M magic -a addr -e ep -n name -d data_file[:data_file...] image\n"
 		"          -A ==> set architecture to 'arch'\n"
 		"          -O ==> set operating system to 'os'\n"
 		"          -T ==> set image type to 'type'\n"
 		"          -C ==> set compression type 'comp'\n"
+		"          -M ==> set image magic to 'magic'\n"
 		"          -a ==> set load address to 'addr' (hex)\n"
 		"          -e ==> set entry point to 'ep' (hex)\n"
 		"          -n ==> set image name to 'name'\n"
@@ -143,7 +145,7 @@ static void process_args(int argc, char
 	int opt;
 
 	while ((opt = getopt(argc, argv,
-			     "a:A:b:c:C:d:D:e:Ef:Fk:i:K:ln:N:p:O:rR:qsT:vVx")) != -1) {
+			     "a:A:b:c:C:d:D:e:Ef:Fk:i:K:lM:n:N:p:O:rR:qsT:vVx")) != -1) {
 		switch (opt) {
 		case 'a':
 			params.addr = strtoull(optarg, &ptr, 16);
@@ -221,6 +223,14 @@ static void process_args(int argc, char
 		case 'l':
 			params.lflag = 1;
 			break;
+		case 'M':
+			params.magic = strtoull(optarg, &ptr, 16);
+			if (*ptr) {
+				fprintf(stderr,	"%s: invalid magic %s\n",
+					params.cmdname, optarg);
+				exit(EXIT_FAILURE);
+			}
+			break;
 		case 'n':
 			params.imagename = optarg;
 			break;
--- a/tools/default_image.c
+++ b/tools/default_image.c
@@ -120,7 +120,7 @@ static void image_set_header(void *ptr,
 	}
 
 	/* Build new header */
-	image_set_magic(hdr, IH_MAGIC);
+	image_set_magic(hdr, params->magic);
 	image_set_time(hdr, time);
 	image_set_size(hdr, imagesize);
 	image_set_load(hdr, addr);
--- a/tools/imagetool.h
+++ b/tools/imagetool.h
@@ -53,6 +53,7 @@ struct image_tool_params {
 	int arch;
 	int type;
 	int comp;
+	unsigned int magic;
 	char *dtc;
 	unsigned int addr;
 	unsigned int ep;
